build/cloudbuild-automation/README.md

```diff
@@ -0,0 +1,67 @@
+- Set the following environment variables 
+    - `PROJECT_ID` with the GCP project ID in which you will submit cloud builds.
+    - `GOOGLE_APPLICATION_CREDENTIALS` with the path to where your GCP credentials are present, most likely, it will be something like `${HOME}/.config/gcloud/application_default_credentials.json`.
+    - You can set them by executing  
+        ```bash
+        export PROJECT_ID=<your-project-id>
+        # This is where the default credentials live, might be different in case your default location is something else
+        export GOOGLE_APPLICATION_CREDENTIALS=${HOME}/.config/gcloud/application_default_credentials.json
+        ```
+
+#### Repository Structure - Cloudbuild-Automation
+
+1. `tf/main.tf` - folder containing Terraform files that will create necessary resources in your GCP account. 
+2. `tf/setup-build-automation.sh` - convenience shell script that runs the terraform script to generate the resources in GCP account. 
+3. `builder-config.yaml` - configuration defining how to build the custom opentelemetry collector. 
+4. `cloudbuild.yaml` - the YAML file containing steps that need to be run in cloud build. The cloudbuild trigger uses this file to build the docker image for the opentelemetry custom collector.
+
+#### How to run this sample
+
+1. `cd` into the `cloudbuild-automation/tf`. 
+2. Modify the variables declared at the top in `setup-build-automation.sh` script to suit your needs. 
+3. Run the `setup-build-automation.sh` script. 
+    ```
+    sh setup-build-automation.sh
+    ```
+4. Upon running the script, terraform should initialize and will generate a few files in the current directory.
+5. **IMPORTANT:** When you run the script for the first time, you will notice that the script fails with an error asking you to connect the repository to your GCP account. You will have to resolve this error before continuing, look at the [Connecting your repository to GCP account](#connecting-your-repository-to-gcp-account) section. 
+6. After running the script successfuly and connecting your repository, you should have a Cloud Build trigger and an Artifact Registry configured for your selected project. You can verify this by looking these services in [Google Cloud Console](https://console.cloud.google.com).
+
+If your Google Cloud project has the required resources - you're done! They should already be configured to automatically build docker images of the collector whenever there is a `push` to the configured branch of your repository - You can test this by making a commit and pushing it. 
+
+#### Notes 
+ - Whenever you make changes to the terraform file to modify/reconfigure the Google cloud resources, you will need to run the `setup-build-automation.sh` script again so that the changes can be put into effect. 
+ - You only need to connect your repository to your GCP account once, unless you need to connect a new repository which was not connected before.
+
+#### Connecting your repository to GCP account
```

Comment:
is there any other way to set this up prior to running the script? It's odd that the user has to get a failure and run the script again

Reply:
Yes, there are resources mentioned in the links that have detailed steps going through the connection process. 
Practically, I found the failure process a little easier since it gives you a URL that takes you to the exact place where you need to connect the repository. 

But yes, it does seem a little odd that a failure is expected.